### ISSUE TITLE
WasmOffsetConverter not initialized when PROXY_TO_PTHREAD

### ISSUE THREAD

**Issue by trybka** [emscripten-core/emscripten#14642]
The `wasmOffsetConverter` object is reconstructed with resetPrototype, but it is incomplete because the incoming object in the worker is undefined. 

Testcase:

stack.cc:
```
#include <emscripten.h>
#include <iostream>

int __attribute__((noinline)) test() {
  return EM_ASM_INT({
    return wasmOffsetConverter.func_starts.length});
}

int main() {
  return test();
}
```
Compile `-O2`
```
em++ -c stack.cc -sUSE_PTHREADS=1 --profiling-funcs -o stack.o -O2
```

Link `-O0` `PROXY_TO_PTHREAD`
```
em++ -o stack.html stack.o -sPROXY_TO_PTHREAD=1 -sUSE_OFFSET_CONVERTER=1 -sUSE_PTHREADS=1 -sEXIT_RUNTIME=1 --profiling-funcs -sPTHREAD_POOL_SIZE=32
```

Fire up webserver and load in Chrome.
`python3 -m http.server 8080`

I get an issue reading `func_starts.length` because it is undefined.
The `wasmOffsetConverter` being fed into the `load` command is undefined. I suspect some kind of ordering issue?

**Comment by kripken** [comment]
As I race, I can't be sure, but I ran it 10 times and it passed for me. I suspect perhaps the recent landing of #14611 may have fixed it? Though the symptom is different, the core issue of a race causing the WasmOffsetConverter to be sent before it is filled out sounds like it could be the same (and was fixed).

**Comment by trybka** [comment]
Yeah, it looks like this was fixed with #14611 !

Awesome.

**Comment by sbc100** [comment]
I believe #14755 should also fix it by removing `wasmOffsetConverter` completely from worker.js.
